src/python_testing/TC_BRBINFO_4_1.py

```diff
@@ -0,0 +1,278 @@
+#    limitations under the License.
+#
+
+# See https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md#defining-the-ci-test-arguments
+# for details about the block below.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs: run1
+# test-runner-run/run1/app: ${ALL_CLUSTERS_APP}
+# test-runner-run/run1/factoryreset: True
+# test-runner-run/run1/quiet: True
+# test-runner-run/run1/app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+# test-runner-run/run1/script-args: --storage-path admin_storage.json --commissioning-method on-network --discriminator 1234 --passcode 20202021 --PICS src/app/tests/suites/certification/ci-pics-values --trace-to json:${TRACE_TEST_JSON}.json --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+# === END CI TEST ARGUMENTS ===
+
+import logging
+import os
+import queue
+import pathlib
+import uuid
+import subprocess
+import time
+import signal
+
+
+import chip.clusters as Clusters
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main, SimpleEventCallback
+from mobly import asserts
+
+from chip import ChipDeviceCtrl  # Needed before chip.FabricAdmin
+
+logger = logging.getLogger(__name__)
+kRootEndpointId = 0
+kICDMEndpointId = 0
+kICDBridgedEndpointId = 2
+kMaxUserActiveModeBitmap = 0x1FFFF
+kMaxUserActiveModeTriggerInstructionByteLength = 128
+
+#uat = cluster.Bitmaps.UserActiveModeTriggerBitmap
+#modes = cluster.Enums.OperatingModeEnum
+#features = cluster.Bitmaps.Feature
+
+class TC_BRBINFO_4_1(MatterBaseTest):
+
+    #
+    # Class Helper functions
+    #
+
+    async def _read_attribute_expect_success(self, endpoint, cluster, attribute, node_id):
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute, node_id=node_id)
+
+    #
+    # Test Harness Helpers
+    #
+
+    def desc_TC_BRBINFO_4_1(self) -> str:
+        """Returns a description of this test"""
+        return "[TC_BRBINFO_4_1] Verification of KeepActive Command [DUT-Server]"
+
+    def steps_TC_BRBINFO_4_1(self) -> list[TestStep]:
+        steps = [
+            TestStep("0", "Preconditions"),
+            TestStep("1a", "TH reads from the ICD the A_IDLE_MODE_DURATION, A_ACTIVE_MODE_DURATION, and ACTIVE_MODE_THRESHOLD attributes"),
+            TestStep("1b", "Simple KeepActive command w/ subscription. ActiveChanged event received by TH contains PromisedActiveDuration"),
+            TestStep("2", "Sends 3x KeepActive commands w/ subscription. ActiveChanged event received ONCE and contains PromisedActiveDuration"),
+            TestStep("3", "KeepActive not returned after 60 minutes of offline ICD"),
+        ]
+        return steps
+
+
+    def pics_TC_BRBINFO_4_1(self) -> list[str]:
+        """ This function returns a list of PICS for this test case that must be True for the test to be run"""
+        pics = [
+            "BRBINFO.C.80",
```

Comment:
```suggestion
            "BRBINFO.S.C80.Rsp",
```

Reply:
But actually, these endpoints are dynamic, so they won't appear in the PICS file since you don't know a priori what the endpoint is. Suggest instead using the @per_endpoint_test decorator to have the test harness select the appropriate endpoints at run time.

Reply:
@cecille 
I don't think ` @per_endpoint_test` works here. The endpoint that you are targeting gets added in the middle of the test and should be identified

Would it make sense for this PICS to be on `MCORE.FS`?

Reply:
Yeah, if the endpoint doesn't appear until after the test is started, then MCORE.FS is appropriate.

Reply:
ack